Activities/GameActivity.cpp

```diff
@@ -372,6 +372,35 @@ void GameActivity::SetCPUTeam(int team)
 */
 }
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+bool GameActivity::IsBuyGUIOpen(int which) {
+    if (which == -1) {
+        for (short player = Players::PlayerOne; player < this->GetPlayerCount(); player++) {
+            if (this->GetBuyGUI(player)->IsEnabled()) {
+                return true;
+            }
+        }
+        return false;
+    }
+    return this->GetBuyGUI(which)->IsEnabled();
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+bool GameActivity::IsBuyGUIVisible(int which) {
```

Comment:
Rename this to `IsBuyGUIOpen` and declare the method as `const`.

Reply:
That's a different function. `Visible` includes the open states and the transition as it closes, while Open only includes `Open` and `Opening`. I chose to use `IsVisible` so that an accidental doubletap of the esc key while the menu was open wouldn't pause the game, it'd have to wait till the menu receded. As in the comment above, I was trying to match the similar methods.

Reply:
I've left it as `IsBuyMenuGUIVisible` to match the method it calls and avoid confusion with the `BuyMenuGUI` method of similar name.